docs/dg/learningBasics.md

```diff
@@ -0,0 +1,140 @@
+
+<!-- ==================================================================================================== -->
+
+## front-end
+
+<box type="info" seamless>
+
+This section is for developers who want to contribute to the back-end of RepoSense. You may skip this section if you want to contribute as a pure front-end developer.
+</box>
+
+The front-end implementation of RepoSense is located in `frontend/src`.
+
+{{ step(1) }} Learn the necessary tools
+
+It is necessary for you to learn the basics of `Vue.js`, `pug`, and `scss` before working on the project. 
+
+  {{ embed('Appendix: tools to learn for front-end development', 'tools.md', level=2) }}
+
+{{ step(3) }} Learn the RepoSense front-end architecture
+
+* You may want to refer to the [front-end architecture](report.html#report-architecture) to understand the relationship
+* Another way for you to understand the front is to use **Vue.js devtools** to inspect the how the various Vue Components interact with each other. You can refer to the [front-end debugging guide](workflow.html#debugging-front-end) for more information.
+
+{{ step(4) }} Get started with contributing
+
+Here is a small [task](https://github.com/reposense/RepoSense/issues/1268) for you to get started on contributing to RepoSense front-end.
+
+* The general goal is to eliminate the passing of variables that can be shared across different components, and move these variables to the **Vuex store**.
+* You can try to move `minDate`, `maxDate`, and `filterSearch` to **Vuex store**. 
+* When in doubt, you can refer to the pull request [#1319](https://github.com/reposense/RepoSense/pull/1319) and [#1320](https://github.com/reposense/RepoSense/pull/1320) to view the implementation detail.
+
+{{ step(4) }} Familiarize yourself with the workflow
+
+1. The next step is to verify that your changes passes the testing and then submit the pull request. You can refer to [the _Workflow_ page](workflow.html).
+
+<!-- ==================================================================================================== -->
+
+## Writing documentation
+
+If you want to make contribution to the RepoSense documentation, you can refer to the guide below.
+
+**This project uses [MarkBind](https://markbind.org/)** for documentation. Follow [this tutorial](https://se-education.org/guides/tutorials/markbind.html) to learn how to use MarkBind for updating project documentation.
+
+**To show some content only in the <tooltip content="i.e., https://reposense.org">production website</tooltip>**, add the `tags="production"` attribute to the HTML element enclosing the content. Similarly, `tags="dev"` will make the content appear only in the <tooltip content="i.e., https://reposense.org/RepoSense">dev website</tooltip>.
+
+```html
+<span tags="production">This will appear in the production website only.</span>
+<span tags="dev">This will appear in the dev website only.</span>
+This will appear in both sites.
+```
```

Comment:
I think this part should not be part of the basics page, since most (if not all) documentation should appear in both sites. Perhaps move it to other page that is more relevant for senior developers?